c++/src/Statistics.hh

```diff
@@ -605,6 +676,43 @@ namespace orc {
     }
   };
 
+  class StripeStatisticsImpl: public StripeStatistics {
+  private:
+    std::unique_ptr<StatisticsImpl> columnStats;
+    std::vector<std::vector<std::unique_ptr<const ColumnStatistics> > > rowIndexStats;
```

Comment:
If using `unique_ptr` is your concern, I believe this is handled by the ORC Adaptor header.
`./c++/src/Adaptor.hh:86:  #define unique_ptr auto_ptr`